presto-main/src/main/java/com/facebook/presto/sql/planner/plan/LateralJoinNode.java

```diff
@@ -36,28 +36,22 @@
 public class LateralJoinNode
         extends PlanNode
 {
-    public enum Type
-    {
-        INNER,
-        LEFT
-    }
-
     private final PlanNode input;
     private final PlanNode subquery;
 
     /**
      * Correlation symbols, returned from input (outer plan) used in subquery (inner plan)
      */
     private final List<Symbol> correlation;
-    private final Type type;
+    private final JoinType type;
 
     @JsonCreator
     public LateralJoinNode(
             @JsonProperty("id") PlanNodeId id,
             @JsonProperty("input") PlanNode input,
             @JsonProperty("subquery") PlanNode subquery,
             @JsonProperty("correlation") List<Symbol> correlation,
-            @JsonProperty("type") Type type)
+            @JsonProperty("type") JoinType type)
```

Comment:
Are all values of `JoinType` valid for `LateralJoinNode`?
If not add a precondition check.